utils/buffer/examples/basic_buffer_pool.rs

```diff
@@ -0,0 +1,41 @@
+// # Simple `BufferPool` Usage
+//
+// This example showcases how to:
+// 1. Creating a `BufferPool`.
+// 2. Obtaining a writable buffer.
+// 3. Writing data into the buffer.
+// 4. Retrieving the data as a referenced slice.
+// 5. Retrieving the data as an owned slice.
+//
+// # Run
+//
+// ```
+// cargo run --example basic_buffer_pool
+// ```
+
+use buffer_sv2::{Buffer, BufferPool};
+
+fn main() {
+    // Create a new BufferPool with a capacity of 32 bytes
```

Comment:
this comment feels a bit confusing/misleading

by reading it with naive eyes, my first thought is: "this buffer pool can only fit 32 bytes"

which is not true!

AFAIU, every buffer pool has exactly 8 slots of variable size (which is 32 bytes here)

Reply:
I think it is 32 bytes, with each slot holding 4 bytes.

```
 BufferPool { pool_back: PoolBack { back_start: 0, len: 0 }, mode: Back, shared_state: SharedState(0), inner_memory: InnerMemory { pool: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], raw_offset: 0, raw_len: 0, slots: [(0, 0), (0, 0), (0, 0), (0, 0), (0, 0), (0, 0), (0, 0), (0, 0)], len: 0 }, system_memory: BufferFromSystemMemory { inner: [], cursor: 0, start: 0 }, start: 0 }
```

Reply:
We have 8 slots with a total capacity of 32 bytes. The slots do not need to have equal capacity, such as 32/8 = 4 bytes each. They can vary in length, as long as the total size remains 32 bytes.

Reply:
I see, my original comment was wrong then.